rust-spec/lightclient/detection/detection.md

```diff
@@ -54,39 +72,55 @@ project.
 
 The light client maintains a simple address book containing addresses
 of full nodes that it can pick as primary and secondaries.  To obtain
-a new header, the light client first does [verification][verification]
-with the primary, and then cross-checks the header with the
-secondaries using this specification.
-
-### Tendermint Consensus and Forks
+a new light block, the light client first does
+[verification](verification) with the primary, and then cross-checks
+the light block (and the trace of light blocks that led to it) with
+the secondaries using this specification.
+
+## Tendermint Consensus and Light Client Attacks
+
+In this section we will give some mathematical definitions of what we
+mean by light client attacks (that are considered in this
+specification) and how they differ from main-chain forks. To this end
+we start by defining some properties of the sequence of blocks that is
+decided upon by Tendermint consensus in normal operation (if the
+Tendermint failure model holds
+[[TMBC-FM-2THIRDS]](TMBC-FM-2THIRDS-link)),
+and then define different
+deviations that correspond to attack scenarios.
 
 #### **[TMBC-GENESIS.1]**
 
 Let *Genesis* be the agreed-upon initial block (file).
 
-#### **[TMBC-FUNC.1]**
+#### **[TMBC-FUNC-SIGN.1]**
 
-> **TODO** be more precise. +2/3 of b.NextV = c.Val signed c. For now
-> the following will do:
+Let *b* and *c* be two light blocks with *b.Header.Height + 1 =
+c.Header.Height*. We define the predicate **signs(b,c)** to hold
+iff *c.Header.LastCommit* is in *PossibleCommit(b)*.
+[[TMBC-SOUND-DISTR-POSS-COMMIT.1]](TMBC-SOUND-DISTR-POSS-COMMIT-link).
+ 
+#### **[TMBC-FUNC-SUPPORT.1]**
 
-Let b and c be two light blocks
-with *b.Header.Height + 1 = c.Header.Height*. We define **signs(b,c)**
-    iff `ValidAndVerified(b, c)`
+Let *b* and *c* be two light blocks. We define the predicate
+**supports(b,c,t)** to hold iff
 
-> **TODO** be more precise. +1/3 of b.NextV signed c. For now
-> the following will do:
+- *t - trustingPeriod < b.Header.Time < t*
+- the voting power in *b.NextValidators* of nodes in *c.Commit*
+  is more than 1/3 of *TotalVotingPower(b.Header.NextValidators)*
 
-Let *b* and *c* be two light blocks. We define **supports(b,c,t)**
-    iff `ValidAndVerified(b, c)` at time *t*
+> That is, if the [Tendermint failure model](TMBC-FM-2THIRDS-link)
+> holds, then *c* has been signed by at least on correct full node, cf.
```

Comment:
Do we need to add an additional assumptions that b comes from Tendermint chain?

Reply:
No. I also will use that to define a verification trace which does not necessarily mean that the b is from the chain. This will come later when `supports` is used in other definitions.